docker.md

```diff
@@ -1,28 +1,41 @@
-# Dockerfile provided by MP
+# Dockerfile conventions
 
-MP should provide a
-[Dockerfile](https://docs.docker.com/engine/reference/builder/) that should
-result in a container which bundles the [tools developed by the MP](cli) along
-with all requirements.
+OCR-D [modules](glossary#ocr-d-module) SHOULD provide
+a [Dockerfile](https://docs.docker.com/engine/reference/builder/)
+that results in containers which bundle the [processor tools](cli)
+along with all requirements.
 
-## Based on ocrd:core
+## Based on OCR-D/core
 
-Docker containers should be based on the [ocrd base
-image](https://hub.docker.com/r/ocrd/core/) which itself is based on Ubuntu
-18.04. For one, this allows MP to use the `ocrd` tool to handle recurrent tasks
-in a spec-conformant way. Besides, it locally installed and containerized
-[CLI](cli) interchangeable.
+Docker images SHOULD be based on the [OCR-D base image](https://github.com/OCR-D/core/pkgs/container/core):
+```Dockerfile
+FROM ghcr.io/ocr-d/core
+```
+(That stage itself is [based on](https://github.com/OCR-D/core/blob/77a385cef8c9dfefda841cb505cc829137ee0578/Makefile#L52) Ubuntu 20.04.)
+
+– or, for CUDA-enabled tools –
+```Dockerfile
+FROM ghcr.io/ocr-d/core-cuda
+```
+(That stage itself is [based on]([https://github.com/OCR-D/core/blob/77a385cef8c9dfefda841cb505cc829137ee0578/Makefile#L52](https://github.com/OCR-D/core/blob/77a385cef8c9dfefda841cb505cc829137ee0578/Makefile#L222)) Ubuntu 20.04 with multiple versions of the Nvidia CUDA runtime.)
+
+This allows using the [`ocrd` multi-purpose tool](https://ocr-d.de/core/api/ocrd/ocrd.cli.html)
+and the [OCR-D/core framework](https://ocr-d.de/core) (with a Python API and a bash library
+to facilitate implementation of new and integration of existing tools)
+to handle recurrent tasks in a spec-conformant way.
+
+Moreover, this makes using natively installed and containerized [CLI](cli) interchangeable.
 
 ## Naming images
 
-Image tags MUST be the same as the project name but with underscore (`_`)
+Docker image tags MUST be the same as the project name but with underscore (`_`)
 replaced with forward slash (`/`).
```

Comment:
```suggestion
replaced with forward slash (`/`) and `ocr-d` spelled in lower case.
```

Reply:
Is that the right context to prescribe `ocr-d` though? IIUC this spec applies to any external module, too. Within github.com/OCR-D we will use that prefix sure, but others should be free to do their thing (including Dockerhub or GHCR or some Gitlab or whatever registry), right?

Reply:
Sure, I only wanted to prevent people from running into the same problem with `ocrd vs OCR-D vs ocr-d`. In DockerHub I could add people to the team "ocrd", so they could publish their repos (which did not need to be in the `OCR-D` organization in GH) to `docker.io/ocrd/calamari` for example.

Since that does not work with ghcr.io AFAIK, we should maybe not restrict the naming of the image at all?

Reply:
> because that's forbidden, see `docker-image-tag(1)`

Actually, it's not forbidden to use underscores in image tags, only in the host name part of the registry. (I did not catch this when I first read it, because the documentation is a bit cryptic.)

And because for external projects (outside of github.com/OCR-D), their connection to OCR-D is _only_ expressed with the typical `ocrd_` prefix, I would even say we should remove this formulation entirely, or at least soften it from a `MUST` to a `SHOULD`. 

For example, it does not make sense to remove the `ocrd_` from [bertsky/ocrd_detectron2](https://github.com/bertsky/ocrd_detectron2/): `bertsky/detectron2` would be confused with a variant of [detectron2](https://github.com/facebookresearch/detectron2) proper, and `bertsky/ocrd/detectron2` is not allowed.

(I am not saying we should completely reverse course and even use `ocrd/ocrd_tesserocr`, although I would not object to it. But mandating the removal of `ocrd` and underscores just does not make sense for external modules.)